From 98f9a1d24442399db3b1b6190010855804ca15e2 Mon Sep 17 00:00:00 2001
From: Oliver Hartkopp <oliver@hartkopp.net>
Date: Fri, 27 Mar 2009 11:21:29 -0700
Subject: [PATCH 09/23] Add support Controller Area Network

t's not a big problem, but it makes a better show in 'ip link show' on
CAN interfaces :-)

I also moved __PF(CAN,can) in ll_proto.c to the same position where it
can be found in if_ether.h .

The only thing i did not know if the __PF(CAN,can) in ll_types.c needs
to be put in #ifdef ARPHDR_CAN like __PF(HWX25,hwx25) is or not. You
definitely know that better than me.
---
 ip/ipaddress.c |    1 +
 lib/ll_proto.c |    2 +-
 lib/ll_types.c |    1 +
 3 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/ip/ipaddress.c b/ip/ipaddress.c
index b055b68..267ecb3 100644
--- a/ip/ipaddress.c
+++ b/ip/ipaddress.c
@@ -109,6 +109,7 @@ void print_link_flags(FILE *fp, unsigned flags, unsigned mdown)
 	_PF(UP);
 	_PF(LOWER_UP);
 	_PF(DORMANT);
+	_PF(ECHO);
 #undef _PF
         if (flags)
 		fprintf(fp, "%x", flags);
diff --git a/lib/ll_proto.c b/lib/ll_proto.c
index 1057ee3..cdd1602 100644
--- a/lib/ll_proto.c
+++ b/lib/ll_proto.c
@@ -84,6 +84,7 @@ __PF(DDCMP,ddcmp)
 __PF(WAN_PPP,wan_ppp)
 __PF(PPP_MP,ppp_mp)
 __PF(LOCALTALK,localtalk)
+__PF(CAN,can)
 __PF(PPPTALK,ppptalk)
 __PF(TR_802_2,tr_802_2)
 __PF(MOBITEX,mobitex)
@@ -94,7 +95,6 @@ __PF(ECONET,econet)
 #endif
 __PF(TIPC,tipc)
 __PF(AOE,aoe)
-__PF(CAN,can)
 
 { 0x8100, "802.1Q" },
 { ETH_P_IP, "ipv4" },
diff --git a/lib/ll_types.c b/lib/ll_types.c
index d47dee7..9d75bd2 100644
--- a/lib/ll_types.c
+++ b/lib/ll_types.c
@@ -70,6 +70,7 @@ __PF(X25,x25)
 #ifdef ARPHRD_HWX25
 __PF(HWX25,hwx25)
 #endif
+__PF(CAN,can)
 __PF(PPP,ppp)
 __PF(HDLC,hdlc)
 __PF(LAPB,lapb)
-- 
1.6.3.3

